### ISSUE TITLE
Migrating User Data from Identity 2.2.1 to 3.x

### ISSUE THREAD

**Issue by fabich** [aspnet/Identity#1102]
Hey all
Is it possible to migrate data from Identity 2.2.1 to 3.x?

I found the following discussion, not sure what current status is.
https://github.com/aspnet/EntityFramework/issues/2167

Migrating only the passwords would be enough for now if complete migration doesn't work.

Thanks for your help

**Comment by HaoK** [comment]
I believe it should just work, @divega any known issues with migrating EF6 dbs to EFCore?

**Comment by HaoK** [comment]
Also see https://github.com/aspnet/Identity/issues/682 for more details:

Specifically the flow that should work is:

Run Enable-Migrations/ensure that an initial migration is created (for your existing 2.2.1 schema)
Install AspNet.Identity.CoreCompat and include the compat namespace to rebase the Identity POCO
Run migrations using EF6 to update schema to V3

Then Identity 3 should be able to read the new schema

**Comment by fabich** [comment]
@HaoK does that imply that I have to use the old PasswordHasher?
services.Configure<PasswordHasherOptions>(options => options.CompatibilityMode = PasswordHasherCompatibilityMode.IdentityV2);

**Comment by blowdart** [comment]
@fabich No, the new one understands both formats, and will even go as far as rehashing to the more secure hash algorithm upon login.

**Comment by xrkolovos** [comment]
I tryied this with my data and it didn't seems to work? Is there any guidance about this?
